src/scene/vcPOI.cpp

```diff
@@ -31,11 +31,300 @@ static const char *vcFRIMStrings[] =
+
+  //TODO Should we allow measure mode to act on a closed polygon?
+  shouldChange = shouldChange || m_pParent->m_line.closed;
+
+  if (shouldChange)
+  {
+    //We need to notify the program state that we have stopped measuring.
+    //Ideally we shouldn't be changing program state here; data should flow in one direction (into this object for example).
+    //But this would require other changes to the program framework.
+    if (pProgramState->activeTool == vcActiveTool_MeasureLine)
+      pProgramState->activeTool = vcActiveTool_Select;
+
+    //We need to remove the preview point
+    if (m_pParent->m_line.numPoints > 0)
+      m_pParent->RemovePoint(pProgramState, m_pParent->m_line.numPoints - 1);
+
+    // TODO: 1452
+    if (m_pParent->m_line.fenceMode != vcRRVM_ScreenLine && m_pParent->m_pFence != nullptr)
+    {
+      vcFenceRenderer_ClearPoints(m_pParent->m_pFence);
+      vcFenceRenderer_AddPoints(m_pParent->m_pFence, m_pParent->m_line.pPoints, m_pParent->m_line.numPoints - 1, m_pParent->m_line.closed);
+    }
+
+    m_pParent->ChangeProjection(pProgramState->gis.zone);
+    if (pProgramState->activeTool == vcActiveTool_MeasureArea)
+      return new vcPOIState_MeasureArea(m_pParent);
+    else
+      return new vcPOIState_General(m_pParent);
+  }
+
+  return this;
+}
+
+//TODO This is very similar to vcPOIState_MeasureLine::ChangeState(). Consider refactoring.
+vcPOIState_General *vcPOIState_MeasureArea::ChangeState(vcState *pProgramState)
+{
+  bool shouldChange = pProgramState->activeTool != vcActiveTool_MeasureArea;
+
+  if (shouldChange)
+  {
+    if (pProgramState->activeTool == vcActiveTool_MeasureArea)
+      pProgramState->activeTool = vcActiveTool_Select;
+
+    //We need to remove the preview point
+    if (m_pParent->m_line.numPoints > 0)
+      m_pParent->RemovePoint(pProgramState, m_pParent->m_line.numPoints - 1);
+
+    // TODO: 1452
+    if (m_pParent->m_line.fenceMode != vcRRVM_ScreenLine && m_pParent->m_pFence != nullptr)
+    {
+      vcFenceRenderer_ClearPoints(m_pParent->m_pFence);
+      vcFenceRenderer_AddPoints(m_pParent->m_pFence, m_pParent->m_line.pPoints, m_pParent->m_line.numPoints - 1, m_pParent->m_line.closed);
+    }
+
+    m_pParent->ChangeProjection(pProgramState->gis.zone);
+    if (pProgramState->activeTool == vcActiveTool_MeasureLine)
+      return new vcPOIState_MeasureLine(m_pParent);
+    else
+      return new vcPOIState_General(m_pParent);
+  }
+
+  return this;
+}
+
+//----------------------------------------------------------------------------------------------------
+// vcPOI
+//----------------------------------------------------------------------------------------------------
+
 vcPOI::vcPOI(vdkProject *pProject, vdkProjectNode *pNode, vcState *pProgramState) :
   vcSceneItem(pProject, pNode, pProgramState)
+  , m_pState(nullptr)
 {
-  m_nameColour = 0xFFFFFFFF;
-  m_backColour = 0x7F000000;
+  m_nameColour = vcIGSW_ImGuiToBGRA(pProgramState->settings.tools.label.textColour);
```

Comment:
The `vcPOI` constructor shouldn't be assigning anything non-const to the values- the `OnNodeUpdate` function can default to user settings if required. None of these values will be saved otherwise.

Reply:
Roger